functions/src/optimize-images.js

```diff
@@ -0,0 +1,73 @@
+import { storage } from 'firebase-functions';
+import Storage from '@google-cloud/storage';
+import { spawn } from 'child-process-promise';
+import mkdirp from 'mkdirp-promise';
+import path from 'path';
+import os from 'os';
+import fs from 'fs';
+
+const gcs = new Storage();
+
+exports.optimizeImages = storage.object().onChange(({ data }) => {
+  const { contentType, resourceState, metageneration } = data;
+  // Exit if this is triggered on a file that is not an image.
+  if (!contentType.startsWith('image/')) {
+    console.log('This is not an image.');
```

Comment:
Do you think we need this logging? We'll get a lot of mess in functions' logging

Reply:
It helps us to understand why a function has finished. For example, I have blurred these messages:
![image](https://user-images.githubusercontent.com/2954281/36104147-dbecd4ea-1019-11e8-8e31-54d6a6b80c43.png)